polkadot/node/subsystem-bench/src/lib/network.rs

```diff
@@ -474,19 +474,17 @@ impl EmulatedPeer {
 	pub async fn send_message(&mut self, message: NetworkMessage) {
 		self.tx_limiter.reap(message.size()).await;
 
-		if self.latency_ms == 0 {
-			self.to_node.unbounded_send(message).expect("Sending to the node never fails");
-		} else {
-			let to_node = self.to_node.clone();
-			let latency_ms = std::time::Duration::from_millis(self.latency_ms as u64);
-
-			// Emulate RTT latency
-			self.spawn_handle
-				.spawn("peer-latency-emulator", "test-environment", async move {
+		let to_node = self.to_node.clone();
+		let latency_ms = std::time::Duration::from_millis(self.latency_ms as u64);
+
+		self.spawn_handle
+			.spawn("peer-latency-emulator", "test-environment", async move {
```

Comment:
I'm not sure I understand how this should help, we would actually get scheduling noise even if latency is zero, so this change would work against making the test more deterministic by setting latency to None.

Reply:
I have the same question. How much of an improvement are we seeing and do we have any clue why that is ?

Reply:
You are right, I used to receive weird results with sending without spawning, but now checked again and everything is ok. I suppose I just had some "noise" on my laptop because of rust-analyzer.

Reply:
I would use a VM with ref hw specs to run the tests and establish gold values for the asserts.

Reply:
Yep, I reverted that